src/mumble/CustomElements.cpp

```diff
@@ -180,43 +180,51 @@ void ChatbarTextEdit::insertFromMimeData(const QMimeData *source) {
 }
 
 bool ChatbarTextEdit::sendImagesFromMimeData(const QMimeData *source) {
-	if (source->hasImage()) {
-		// Process the image pasted onto the chatbar.
-		if (g.bAllowHTML) {
+	if (g.bAllowHTML) {
+		if (source->hasImage()) {
+			// Process the image pasted onto the chatbar.
 			QImage image = qvariant_cast< QImage >(source->imageData());
-
-			QString imgHtml = QLatin1String("<br />") + Log::imageToImg(image);
-
-			if ((g.uiImageLength == 0) || static_cast< unsigned int >(imgHtml.length()) < g.uiImageLength) {
-				emit pastedImage(imgHtml);
+			if (emitPastedImage(image)) {
 				return true;
 			} else {
 				g.l->log(Log::Information, tr("Unable to send image: too large."));
 			}
-		}
-	} else if (source->hasUrls()) {
-		// Process the files dropped onto the chatbar. URLs here should be understood as the URIs of files.
-		QList< QUrl > urlList = source->urls();
-
-		int count = 0;
-		for (int i = 0; i < urlList.size(); ++i) {
-			QString path = urlList[i].toLocalFile();
-			QImage image(path);
 
-			if (image.isNull())
-				continue;
-
-			QString imgHtml = QLatin1String("<br />") + Log::imageToImg(image);
-
-			if (static_cast< unsigned int >(imgHtml.length()) < g.uiImageLength) {
-				emit pastedImage(imgHtml);
-				++count;
-			} else {
-				g.l->log(Log::Information, tr("Unable to send image %1: too large.").arg(path));
+		} else if (source->hasUrls()) {
+			// Process the files dropped onto the chatbar. URLs here should be understood as the URIs of files.
+			QList< QUrl > urlList = source->urls();
+
+			int count = 0;
+			for (int i = 0; i < urlList.size(); ++i) {
+				QString path = urlList[i].toLocalFile();
+				QImage image(path);
+
+				if (image.isNull())
+					continue;
+				if (emitPastedImage(image)) {
+					++count;
+				} else {
+					g.l->log(Log::Information, tr("Unable to send image %1: too large.").arg(path));
+				}
 			}
+
+			return (count > 0);
 		}
+	}
+	g.l->log(Log::Information, tr("This server does not allow sending images."));
+	return false;
+}
 
-		return (count > 0);
+bool ChatbarTextEdit::emitPastedImage(QImage image) {
+	int maxSize = 65536;
+	if ((g.uiImageLength > 0) || (g.uiImageLength < 65536)) {
```

Comment:
This will always be true. I think you probably wanted to use `&&` here instead of `||`. Furthermore I think you should not use this hardcoded number but the `maxSize` variable instead:
```suggestion
	if ((g.uiImageLength > 0) && (g.uiImageLength < maxSize)) {
```

Reply:
You are right. What was I even thinking?